src/Aspire.Hosting.JavaScript/JavaScriptHostingExtensions.cs

```diff
@@ -586,6 +613,28 @@ private static void AddInstaller<TResource>(IResourceBuilder<TResource> resource
                     .WithWorkingDirectory(resource.Resource.WorkingDirectory)
                     .WithArgs(installCommand.Args);
 
+                // Add a second validation hook for the specific package manager after the command is set
+                installerBuilder.OnBeforeResourceStarted(async (installerResource, e, ct) =>
+                {
+                    // Check which package manager is being used and validate it
+                    var executableName = packageManager.ExecutableName.ToLowerInvariant();
+                    switch (executableName)
+                    {
+                        case "npm":
+                            var npmInstallationManager = e.Services.GetRequiredService<NpmInstallationManager>();
+                            await npmInstallationManager.EnsureInstalledAsync(throwOnFailure: false, ct).ConfigureAwait(false);
+                            break;
+                        case "yarn":
+                            var yarnInstallationManager = e.Services.GetRequiredService<YarnInstallationManager>();
+                            await yarnInstallationManager.EnsureInstalledAsync(throwOnFailure: false, ct).ConfigureAwait(false);
+                            break;
+                        case "pnpm":
+                            var pnpmInstallationManager = e.Services.GetRequiredService<PnpmInstallationManager>();
+                            await pnpmInstallationManager.EnsureInstalledAsync(throwOnFailure: false, ct).ConfigureAwait(false);
+                            break;
+                    }
```

Comment:
No, let's not do it this way.

Why can't we pass the command name into the "InstallationManager"? Then this could be

```
installationManager.EnsureInstalledAsync(packageManager.ExecutableName)
```

or similar